api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -742,6 +743,45 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     return ResponseEntity.ok(response);
   }
 
+  @Override
+  public ResponseEntity<EmptyResponse> renameNotebook(String workspace, String workspaceName, NotebookRename rename) {
```

Comment:
What does this do if you name a notebook the name of something that already exists in the notebooks folder? Do we want to support error messages on name collision?

Reply:
I've addressed this at the UI level.